templates/person/show.json.twig

```diff
@@ -0,0 +1,71 @@
+{
+    "@context": "https://schema.org",
+    "@graph": [
+        {
+            "@type": "Person"
+            {% if person.firstname %},"givenName": "{{ person.firstName}}" {% endif %}
+            {% if person.lastName %},"familyName": "{{ person.lastName}}" {% endif %}
+            ,"gender": "{{ person.gender }}"
+            {% if (person.wikipediaUrl) or (person.viafUrl) %}
+            ,"sameAs": [
+                {% if person.wikipediaUrl %} "{{ person.wikipediaUrl}}"{% endif %}
+                {% if (person.wikipediaUrl) and (person.viafUrl) %} , {% endif %}
+                {% if person.viafUrl %} "{{ person.viafUrl}}" {% endif %}
+            ]
+            {% endif %}
+            {% if person.dob %},"birthDate": "{{person.dob}}" {% endif %}
+            {% if person.dod %}, "deathDate": "{{person.dod}}" {% endif %}
+            {% if person.cityOfBirth %}, "birthPlace": "{{person.cityOfBirth}}" {% endif %}
+            {% if person.cityOfDeath %}, "deathPlace": "{{person.cityOfDeath}}" {% endif %}
+            ,"@id": "{{ url('person_show', {'id':person.id}) }}"
+        }
+        {% for titleRole in person.getTitleRoles("Author") %}
+         {% if titleRole.getRole.getName == "Author" %}
+                {% set book = titleRole.title %}
+                {% if book %}
+                ,{
+                    "@type": "Book",
+                    "@id": "{{ url('title_show', {'id':book.id}) }}" ,
+                    "name": "{{book.title}}"
+                    {% if book.genre %} ,"genre": "{{book.genre}}" {% endif %}
+                    {% if book.pubdate %},"datePublished": "{{book.pubdate}}" {% endif %},
+                    "locationCreated": {
+                        "@type": "Place"
+                        {% if book.locationOfPrinting %}, "name": "{{book.locationOfPrinting}}" {% endif %},
```

Comment:
I think the conditional should probably go a level up:

```twig
{% if book.locationOfPrinting %}
 "locationCreated": {
      "@type": "Place",
      "name": {{ book.locationOfPrinting }}
      {# ... Et cetera ... #}
    }
{% endif %}

```

Reply:
I only have one question. If an address has a location (which I think all of them are geo locations), will it necessarily have the latitude and longitude?